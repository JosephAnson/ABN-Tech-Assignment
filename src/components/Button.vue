<template>
  <a
    class="rounded-md bg-#F5C519 px-4 py-2 text-base font-semibold leading-7 text-primary-900 shadow-sm hover:bg-#BE9917 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-#BE9917"
  >
    <slot />
  </a>
</template>
