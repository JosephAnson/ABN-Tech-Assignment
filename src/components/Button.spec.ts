import { render } from "@testing-library/vue"
import { describe, expect, test } from "vitest"
import Button from "./Button.vue"

describe("Button", () => {
  test("should render button", () => {
    const buttonText = "Test Button"
    const { html, getByText } = render(Button, {
      slots: {
        default: buttonText,
      },
    })
    expect(getByText(buttonText)).toBeTruthy()
    expect(html()).toMatchSnapshot()
  })
})
