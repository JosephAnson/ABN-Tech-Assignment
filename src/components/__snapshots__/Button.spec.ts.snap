// Vitest Snapshot v1

exports[`Button > should render button 1`] = `"<a class=\\"rounded-md bg-#F5C519 px-4 py-2 text-base font-semibold leading-7 text-primary-900 shadow-sm hover:bg-#BE9917 focus-visible:outline focus-visible:outline-2 focus-visible:outline-offset-2 focus-visible:outline-#BE9917\\">Test Button</a>"`;
